documentation/docs/04-advanced-routing.md

```diff
@@ -63,6 +63,12 @@ export function load(event) {
 
 > If you don't handle 404 cases, they will appear in [`handleError`](/docs/hooks#shared-hooks-handleerror)
 
+### Optional parameters
+
+A route like `[lang]/home` contains a parameter named `lang` which is required. Sometimes it's beneficial to make these parameters optional, so that in this example both `home` and `en/home` point to the same page. You can do that by wrapping the parameter in another bracket pair: `[[lang]]/home`
+
+Note that an optional route parameter cannot follow a rest parameter (`[...rest]/[[optional]]`) or vice versa, since this creates ambiguity.
```

Comment:
```suggestion
Note that an optional route parameter cannot follow a rest parameter (`[...rest]/[[optional]]`), since parameters are matched 'greedily' and the optional parameter will always be unused.
```

Reply:
Perhaps "would always be unused"? Because we're not warning people against doing this, we're saying it's disallowed, and so this is what _would_ happen if they were allowed to, not what _will_ happen if they do.